### ISSUE TITLE
Matches style callback should support collections of non-dicts

### ISSUE THREAD

**Issue by stevelacey** [dgilland/pydash#168]
Maybe this is a design decision/limitation of other things already in pydash, but I expected this to work:

```py
users = User.objects.filter(...)
pydash.find(users, dict(first_name="Steve"))
=> None
```

Like it does in fnc:

```py
users = User.objects.filter(...)
fnc.find(dict(first_name="Steve"), users)
=> <User: Steve>
```

I see `matches styles callback` takes a dict, but it only seems to work on lists of dicts, aka it doesn't match on attributes/properties, like I expected, and as per fnc, it would be great if these had parity.

I don't really need fnc and I'd use pydash instead if this worked, the `matches property style` works but isn't a great way to write this, `matches styles callback` requires zero explanation to the uninitiated.

**Comment by dgilland** [comment]
Do you have a minimal, self-contained example that demonstrates the behavior? I can then check to see what's happening under the hood to see how feasible it would be to change. Thanks!

**Comment by stevelacey** [comment]
Sure thing, you can exercise it with a `namedtuple`:

```py
from collections import namedtuple
import pydash

User = namedtuple('User', ['first_name'])

users = [User('Steve')]

pydash.find(users, dict(first_name='Steve'))
=> None
```

```py
from collections import namedtuple
import fnc

User = namedtuple('User', ['first_name'])

users = [User('Steve')]

fnc.find(dict(first_name='Steve'), users)
=> User(first_name='Steve')
```

**Comment by dgilland** [comment]
@stevelacey I've pushed a potential fix for this to the `develop` branch. Can you install it for your application and let me know if it resolves your issue?

**Comment by stevelacey** [comment]
@dgilland yep works perfect 👌🏼

**Comment by dgilland** [comment]
This is released in `5.1.0`: https://pypi.org/project/pydash/5.1.0/
